editor/import/resource_importer_texture.cpp

```diff
@@ -523,18 +525,29 @@ Error ResourceImporterTexture::import(const String &p_source_file, const String
 				int new_width = size_limit;
 				int new_height = target_image->get_height() * new_width / target_image->get_width();
 
-				target_image->resize(new_width, new_height, Image::INTERPOLATE_CUBIC);
+				target_image->resize(new_width, new_height, Image::INTERPOLATE_LANCZOS);
 			} else {
 				int new_height = size_limit;
 				int new_width = target_image->get_width() * new_height / target_image->get_height();
 
-				target_image->resize(new_width, new_height, Image::INTERPOLATE_CUBIC);
+				target_image->resize(new_width, new_height, Image::INTERPOLATE_LANCZOS);
 			}
 
 			if (normal == 1) {
 				target_image->normalize();
 			}
 		}
+		// Apply the resizing.
+		if (!Math::is_equal_approx(process_scale, 1.0f, 0.001f)) {
```

Comment:
Is specifying a custom epsilon needed here? We try to avoid it unless it's strictly required for the comparison to actually work.